### PR TITLE
Homework 3

### PR DESCRIPTION
**Pull request by artemd24**

### CODE REVIEW

File: homework3/task01/task01.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from typing import Callable
+
+
+def cache(times: int):
+    cacher = dict()
+
+    def decorator(func: Callable) -> Callable:
+        def wrapper(*args, **kwargs):
```
Comment (apaniukov) [inline homework3/task01/task01.py:10]: Use `functools.wraps` here.

File: homework3/task01/task01.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            if func.__name__ not in cacher:
+                cacher[func.__name__] = [func(), times]
+            else:
+                if cacher[func.__name__][1] == 0:
+                    del cacher[func.__name__]
+                    cacher[func.__name__] = [func(), times]
```
Comment (apaniukov) [inline homework3/task01/task01.py]: Why do you add the function to the cache again?

File: homework3/task01/task01.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+from typing import Callable
+
+
+def cache(times: int):
```
Comment (apaniukov) [inline homework3/task01/task01.py:5]: This decorator can be used only with functions without arguments. Fix this.

File: homework3/task01/task01.py (hunk 1 above)
Comment (artemd24) [reply homework3/task01/task01.py:10]: Done

File: homework3/task01/task01.py (hunk 2 above)
Comment (artemd24) [reply homework3/task01/task01.py]: Fixed

**Review by apaniukov — APPROVED** [review]

**Review by apaniukov — APPROVED** [review]

**Pull request merged**